### DISCUSSION TITLE
ListBox selection pointerpressed to pointerRelease for DnD

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#9470] [Q&A]
Hi guys, 

I'm using avalonia behaviors for DnD. On that purpose I need to keep selection when I'm dragging. for that, I want to changed the selection from ListBox from pointerpressed to pointerRelease. Is it possible ?

My case with demo avalonia, select three items:

![Case 1](https://user-images.githubusercontent.com/11456289/202725600-c7eff1dd-e87a-4c77-95b3-213c2524ce5d.PNG)

after I'm pressing on Item 4 to start drag : 

![Case 2](https://user-images.githubusercontent.com/11456289/202725715-6c5e9ad5-2ab8-470b-b384-a1550b9d11ab.PNG)

**Comment by maxkatz6** [comment]
You can create a custom control that inherits ListBox.
After it is possible to handle OnPointerPressed without calling base.OnPointerPressed (basically disabling default behavior).
And copy [existing implementation of base.OnPointerPressed](https://github.com/AvaloniaUI/Avalonia/blob/release/0.10.18/src/Avalonia.Controls/ListBox.cs#L149-L164) to OnPointerReleased. 
Should be simple, but haven't tested.

**Reply by ctacke** [reply]
Unfortunately the custom control also does not receive the event.
